### PR TITLE
[LMS-329][Backend] Filter, Sort, and Search API

### PR DESCRIPTION
**Pull request by mikorosales**
## Issue Link
[LMS-329](https://framgiaph.backlog.com/view/LMS-329)

### API filter endpoints

Filter Status:
- /api/course/?is_active={value}
- value = true || false
- ex. url = /api/course/?is_active=true
----
Filter Category (Multiple):
#### Note for handling multiple category filter! this is a CSV, use %2C
- /api/course/?course_category={value}
- value = 1,2,3...
- ex. url = /api/course/?course_category=1%2C2%2C3
----
Search Title:
- /api/course/?title={value}
- value = Course 1
- ex. url = /api/course/?title=Course%201
----
Order Title:
- /api/course/?sort={value}
- value = title_asc || title_desc
- ex. url = /api/course/?sort=title_asc
----
To use all features you can use this endpoint
- /api/course/?sort=&is_active=&course_category=&title=

## Steps to reproduce
1. Docker compose up
2. Make migrations for is_active field
3. http://localhost:8000/api/course/?sort=&is_active=&course_category=&title=
4. Put values, you can base it from the definition of done

## Affected Components / Functionalities / Page
RadioButton
LayoutContainer
ViewAs
useSearchCourse

## Test Cases
It it possible that all your course status is set to TRUE, to test inactive state, do the following:
1. Login to http://localhost:8000/admin/ (UN: admin, PW: password)
2. Course > select courses and set 'is_active' field to false

## Notes
n/a

## Screenshots
image
image

### CODE REVIEW

File: client/src/pages/trainer/courses/filter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        setCategory(category.data);
+      } catch (error) {
+        console.error(error);
+      }
+    }
+    void fetchdata();
+  }, []);
+
+  const handleCategoryChange = (id: number): void => {
+    // console.log('Selected category ID:', id * -1 + 1);
```
Comment (ragg09-sunasterisk) [inline client/src/pages/trainer/courses/filter/index.tsx:47]: remove comments

File: client/src/pages/trainer/courses/filter/index.tsx
Comment (ragg09-sunasterisk) [inline client/src/pages/trainer/courses/filter/index.tsx:1]: can you move the logics in `/shared/hooks/useCourseFilter.ts`

File: client/src/pages/trainer/courses/filter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    const isCourseActive = course.is_active;
+    const shouldIncludeCourse =
+      (selectedStatus && isCourseActive) ||
+      (!selectedStatus && !isCourseActive);
+
+    return isCourseInSelectedCategories && shouldIncludeCourse;
+  });
+
+  return (
+    <>
```
Comment (ragg09-sunasterisk) [inline client/src/pages/trainer/courses/filter/index.tsx:78]: for standard, use <Fragment></Fragment>

**Review by ragg09-sunasterisk — COMMENTED** [review]
Insert the neccessary command in updating the migration in **steps to reproduce** next time
```
python manage.py makemigrations
python manage.py migrate
```

I encounter problem , here is the issue
Filter_error

File: client/src/apis/index.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import axios from 'axios';
 
 const axiosInstance = axios.create({
-  baseURL: process.env.NEXT_PUBLIC_API_BASE_URL
```
Comment (ragg09-sunasterisk) [inline client/src/apis/index.ts:4]: revert this changes
we will use `baseURL: process.env.NEXT_PUBLIC_API_BASE_URL`

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- [x] Filters inactive and active courses.
- [x] Also Filters courses category.
- [x]  No ESLint warnings or errors. 
- [ ] No error found, when running npm run build.

Comments: 5
Test time: 120 mins
Iterations: 2

Proof:
filter
image

**Review by kdacudag — APPROVED** [review]

**Review by Marielleregner7 — APPROVED** [review]

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**

- [x] Able to Perform Filter, Sort, and Search API using Postman.
- [x] No ESLint warnings or errors.
- [x] No error found, when running npm run build.

Comments: 6
Test time: 180 mins
Iterations: 3

Proof:
mikochan
